src/components/program/programPlanAdminModals/PerpetualPlanModal.tsx

```diff
@@ -0,0 +1,298 @@
+            } else if (!values.hasGiftPlan && productGiftPlan?.id) {
+              deleteProductGiftPlan({
+                variables: {
+                  productGiftPlanId: productGiftPlan.id,
+                },
+              })
+                .then(() => {
+                  setLoading(false)
+                  message.success(formatMessage(commonMessages.event.successfullySaved))
+                  onSuccess()
+                  onProductGiftPlanRefetch?.()
+                  onRefetch?.()
+                })
+                .catch(err => console.log(err))
+            } else {
+              setLoading(false)
+              message.success(formatMessage(commonMessages.event.successfullySaved))
+              onSuccess()
+              onProductGiftPlanRefetch?.()
+              onRefetch?.()
+            }
+          })
+      })
+      .catch(() => {})
+  }
+
+  return (
+    <AdminModal
+      title={formatMessage(commonMessages.label.perpetualPlan)}
+      icon={<FileAddOutlined />}
+      footer={null}
+      renderFooter={({ setVisible }) => (
+        <>
+          <Button className="mr-2" onClick={() => setVisible(false)}>
+            {formatMessage(commonMessages.ui.cancel)}
+          </Button>
+          <Button type="primary" loading={loading} onClick={() => handleSubmit(() => setVisible(false))}>
+            {formatMessage(commonMessages.ui.confirm)}
+          </Button>
+        </>
+      )}
+      renderTrigger={({ setVisible }) => {
+        return (
+          renderTrigger?.({
+            onOpen: () => {
+              setVisible(true)
+            },
+            onClose: () => setVisible(false),
+          }) || null
+        )
+      }}
+      {...modalProps}
+    >
+      <Form
+        form={form}
+        layout="vertical"
+        colon={false}
+        hideRequiredMark
+        initialValues={{
+          title: programPlan?.title || '',
+          isPublished: !!programPlan?.publishedAt,
+          isParticipantsVisible: !!programPlan?.isParticipantsVisible,
+          currencyId: programPlan?.currencyId,
+          listPrice: programPlan?.listPrice || 0,
+          sale: programPlan?.soldAt
+            ? {
+                price: programPlan.salePrice,
+                soldAt: programPlan.soldAt,
+                isTimerVisible: !!programPlan?.isCountdownTimerVisible,
+              }
+            : null,
+          type: programPlan?.type || 3,
+          description: BraftEditor.createEditorState(programPlan ? programPlan.description : null),
+          groupBuyingPeople: programPlan?.groupBuyingPeople || 1,
+          hasGiftPlan: productGiftPlan?.giftPlan.id !== undefined ? true : false,
+          productGiftPlanId: productGiftPlan?.id,
+          giftPlanProductId: productGiftPlan?.giftPlan.id || undefined,
+          giftPlanStartedAt: productGiftPlan?.startedAt ? moment(productGiftPlan.startedAt) : '',
+          giftPlanEndedAt: productGiftPlan?.startedAt ? moment(productGiftPlan.endedAt) : '',
```

Comment:
must be productGiftPlan?.EndedAt

Reply:
This code is inherited from previous code.
Okey, I will change it to 'EndedAt'.